cluster/src/dunit/scala/io/snappydata/cluster/SecureQueryRoutingDUnitTest.scala

```diff
@@ -0,0 +1,229 @@
+/*
+ * Copyright (c) 2016 SnappyData, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you
+ * may not use this file except in compliance with the License. You
+ * may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing
+ * permissions and limitations under the License. See accompanying
+ * LICENSE file.
+ */
+
+package io.snappydata.cluster
+
+import java.sql.{Connection, DriverManager, ResultSet}
+
+import com.pivotal.gemfirexd.{Attribute, TestUtil}
+import com.pivotal.gemfirexd.security.LdapTestServer
+import io.snappydata.test.dunit.AvailablePortHelper
+
+import org.apache.spark.Logging
+import org.apache.spark.sql.collection.Utils
+
+class SecureQueryRoutingDUnitTest(val s: String)
+    extends ClusterManagerTestBase(s) with Logging {
+
+  override def setUp(): Unit = {
+    val ldapServer = LdapTestServer.getInstance(getClass.getResource("/auth.ldif").getPath)
+    if (!ldapServer.isServerStarted) {
+      ldapServer.startServer()
+    }
+    setSecurityProps(ldapServer)
+    super.setUp()
+  }
+
+  override def tearDown2(): Unit = {
+    super.tearDown2()
+    val ldapServer = LdapTestServer.getInstance()
+    if (ldapServer.isServerStarted) {
+      ldapServer.stopService()
+    }
+  }
+
+  val ldapAdminUser = "gemfire3"
+  val snappyAdminUser = "gemfire2"
+  val jdbcUser = "gemfire1"
+
+  def setSecurityProps(ldapServer: LdapTestServer): Unit = {
+    import com.pivotal.gemfirexd.Property.{AUTH_LDAP_SERVER, AUTH_LDAP_SEARCH_BASE, AUTH_LDAP_SEARCH_DN, AUTH_LDAP_SEARCH_PW}
+    var props = scala.collection.mutable.Map[String, String]()
+    props += (Attribute.AUTH_PROVIDER ->
+        com.pivotal.gemfirexd.Constants.AUTHENTICATION_PROVIDER_LDAP)
+    props += (AUTH_LDAP_SERVER -> s"ldap://localhost:${ldapServer.getServerPort}/")
+    props += (AUTH_LDAP_SEARCH_BASE -> "ou=ldapTesting,dc=pune,dc=gemstone,dc=com")
+    props += (AUTH_LDAP_SEARCH_DN ->
+        s"uid=$ldapAdminUser,ou=ldapTesting,dc=pune,dc=gemstone,dc=com")
+    props += (AUTH_LDAP_SEARCH_PW -> ldapAdminUser)
+    for ((k, v) <- props) System.setProperty(k, v)
+
+    props += (Attribute.USERNAME_ATTR -> snappyAdminUser)
+    props += (Attribute.PASSWORD_ATTR -> snappyAdminUser)
```

Comment:
Can directly use SecurityTestUtils.startLdapServerAndGetBootProperties as mentioned (locator port is in ClusterManagerTestBase.locPort).